firmware-support/bittide-hal/src/manual_additions/timer.rs

```diff
@@ -26,6 +26,14 @@ use ufmt::uwrite;
 
 pub mod self_test;
 
+pub enum WaitResult {
```

Comment:
This could have `uDebug`/`Debug` derives, might be nice for logging.
```suggestion
use ufmt::derive::uDebug;

#[derive(uDebug, Debug)]
pub enum WaitResult {
```